src/lua/oltp_common.lua

```diff
@@ -270,6 +292,60 @@ CREATE TABLE sbtest%d(
    end
 end
 
+function bulk_inserts(con, table_num)
+
+   if (sysbench.opt.table_size > 0) then
+      time = os.date("*t")
+   end
+   iquery = ""
+
+   if sysbench.opt.auto_inc then
+      iquery = "INSERT INTO sbtest" .. table_num .. "(k, c, pad) VALUES"
+   else
+      iquery = "INSERT INTO sbtest" .. table_num .. "(id, k, c, pad) VALUES"
+   end
+
+   con:bulk_insert_init(iquery)
+
+   local c_val
+   local pad_val
+
+   cursize = 1
+
+
+   for i = 1, sysbench.opt.batch_insert_count do
+
+       if (sysbench.opt.batch_insert_count <=2000) then
+           c_val = get_c_value()
+           pad_val = get_pad_value()
+       else
+           c_val = get_c_value_smaller()
+           pad_val = get_pad_value_smaller()
+       end
+
+       if (sysbench.opt.auto_inc) then
+          query = string.format("(%d, '%s', '%s')",
+                                sysbench.rand.default(1, sysbench.opt.table_size),
+                                c_val, pad_val)
+       else
+          query = string.format("(%d, %d, '%s', '%s')",
+                                sysbench.opt.table_size+pk_number,
+                                sysbench.rand.default(1, sysbench.opt.table_size),
+                                c_val, pad_val)
+          pk_number=pk_number+1
```

Comment:
This will not work in multi threaded scenarios.

Reply:
We need to have the code which we are using in bulk_load. What was the problem with the previous approach of using a number between and 1 and MAX_INT

Reply:
Just read your previous comment. In that case, we can't do bulk insert in its current mode for non auto_inc cases. I could recommend. Throwing an error for this case rather than supporting it partially.